### PR TITLE
add archiveopteryx 3.2.0

### PR DESCRIPTION
**Pull request by phunehehe**
Am I doing this right?
If this works out OK, is there a reason not to include it in 15.09? I think it might be fine because it can't break anything existing.

### CODE REVIEW

File: pkgs/servers/mail/archiveopteryx/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, fetchurl, openssl, perl, zlib }:
+with builtins;
+let
+  version = "3.2.0";
+in stdenv.mkDerivation {
+  name = "${baseNameOf ./.}-${version}";
```
Comment (garbas) [inline pkgs/servers/mail/archiveopteryx/default.nix]: - dont rely on baseNameOf and just type name (this makes expression not depending in which directory it is)
- you can use ``rec`` and drop ``let ... in`` and also use ``${name}`` in the url

Comment (garbas) [inline pkgs/servers/mail/archiveopteryx/default.nix]: - also then ``with builtins;`` is not needed

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: How strong is the preference of `rec` over `let ... in`? I kinda dislike `rec` :)
The rest has been updated as you said.

Comment (jagajaga) [inline pkgs/servers/mail/archiveopteryx/default.nix]: We prefer `rec`.

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: can't argue with that :) updated

File: pkgs/servers/mail/archiveopteryx/default.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    url = "http://archiveopteryx.org/download/${name}.tar.bz2";
+    sha256 = "0i0zg8di8nbh96qnyyr156ikwcsq1w9b2291bazm5whb351flmqx";
+  };
+  buildInputs = [ openssl perl zlib ];
+  installPhase = ''
+    INSTALLROOT=$out make install
+    mkdir $out/bin
+    ln --symbolic $out/usr/local/archiveopteryx/sbin/* $out/bin/
+    ln --symbolic $out/usr/local/archiveopteryx/bin/* $out/bin/
+  '';
```
Comment (jagajaga) [inline pkgs/servers/mail/archiveopteryx/default.nix]: What about `meta` part?

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: Hmm I didn't see any note about a required `meta` part. Can you paste a link or something? I'll add that in a bit.

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: added `meta` by following https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/mail/dovecot-pigeonhole/default.nix#L29-L34

Comment (garbas) [inline pkgs/servers/mail/archiveopteryx/default.nix]: @phunehehe evaluation is failing with https://travis-ci.org/NixOS/nixpkgs/builds/98510983#L155

check for available licenses here https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: woa, strange that it works on my machine...

Comment (phunehehe) [inline pkgs/servers/mail/archiveopteryx/default.nix]: I was a bit lost determining which BSD license this is https://github.com/aox/aox/blob/master/doc/readme/license.txt
But then I found https://github.com/aox/aox/blob/master/doc/readme/COPYING. PostgreSQL License it is.

File: pkgs/servers/mail/archiveopteryx/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  meta = with stdenv.lib; {
+    homepage = http://archiveopteryx.org/;
+    description = ''
```
Comment (jagajaga) [inline pkgs/servers/mail/archiveopteryx/default.nix]: Please omit pkg's name in the description, make it one line and remove dot at the end.

**Comment by vcunat** [comment]
The paths `$out/usr/local/archiveopteryx/{bin,lib,...}` are really ugly. That should get fixed (best before picking to 15.09).

**Comment by jagajaga** [comment]
@vcunat why they are ugly?

**Comment by jagajaga** [comment]
Oh, yes, they are..

**Comment by vcunat** [comment]
@phunehehe: I don't know how to use this package &ndash; would you test e2c260a9b84?

**Comment by phunehehe** [comment]
@vcunat sure I'll test. I use it in [runix-archiveopteryx](https://gitlab.com/phunehehe/runix-archiveopteryx/blob/df0d25186bb70ffdd0bbd7a749bbdff5d893749d/default.nix#L81).

**Comment by vcunat** [comment]
Pushed to master and 15.09.

**Comment by phunehehe** [comment]
:dancers:

**Pull request merged**